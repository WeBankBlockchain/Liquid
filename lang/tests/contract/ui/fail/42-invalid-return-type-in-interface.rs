use liquid_lang as liquid;

#[liquid::interface(name = auto)]
mod foo {
    extern "liquid" {
        fn bar(&self) -> f32;
    }
}

fn main() {}
