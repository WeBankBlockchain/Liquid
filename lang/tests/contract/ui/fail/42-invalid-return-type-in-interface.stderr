error[E0277]: the trait bound `f32: You_Should_Use_An_Valid_Output_Type` is not satisfied
 --> $DIR/42-invalid-return-type-in-interface.rs:6:9
  |
6 |         fn bar(&self) -> f32;
  |         ^^^^^^^^^^^^^^^^^^^^^ the trait `You_Should_Use_An_Valid_Output_Type` is not implemented for `f32`
